### PR TITLE
Update new Wasm gas policy

### LINKED ISSUE #582: Update owasm gas meter to use more than just the default of 1 for all operations
Can use [this](https://ewasm.readthedocs.io/en/mkdocs/determining_wasm_gas_costs/) as a reference.

### PR DESCRIPTION
**Pull request by taobun**
fixed #582

### CODE REVIEW

File: chain/owasm/gas_policy.go [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+func NewDefaultWasmGasPolicy() WasmGasPolicy {
+	return WasmGasPolicy{
+		GasMap: defaultGas,
+	}
+}
+
+func (p *WasmGasPolicy) GetCost(key compiler.Instr) int64 {
+	gasCost, found := p.GasMap[key.Op]
+	if !found {
+		panic(fmt.Sprintf("Cannot found instruction %s", key.Op))
```
Comment (sorawit) [inline chain/owasm/gas_policy.go]: If we panic here. I assume running wasm's Execution function will return error, correct?

File: chain/owasm/gas_policy.go [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+	"f32.convert_s/i64":   3,
+	"f64.convert_s/i32":   3,
+	"f64.convert_s/i64":   3,
+
+	// Type-parametric operators.
+	"drop":   3,
+	"select": 3,
+}
+
+type WasmGasPolicy struct {
```
Comment (sorawit) [inline chain/owasm/gas_policy.go]: Let's call this `BandChainGasPolicy`

File: chain/owasm/gas_policy.go [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+	GasMap map[string]int64
```
Comment (sorawit) [inline chain/owasm/gas_policy.go]: Don't think we need this data structure at all. `GetCost` can just read from the map directly?

**Review by sorawit — CHANGES_REQUESTED** [review]

File: chain/owasm/gas_policy.go (hunk 1 above)
Comment (taobun) [reply chain/owasm/gas_policy.go]: Nope, Good catch! I change to return UnknownGasCost instead.

**Review by sorawit — DISMISSED** [review]

**Review by evilpeach — APPROVED** [review]

**Review by sorawit — APPROVED** [review]

**Pull request merged**